### PR TITLE
`doc` attribute for docstrings

### PR DESCRIPTION
**Pull request by neunenak**
cf. https://github.com/casey/just/issues/1955 , it would be convenient if there was a way to specify a recipe documentation string with an attribute. Also this would make possible to provide a docstring that is different from actual comments in the justfile.

### CODE REVIEW

**Comment by casey** [comment]
I think this needs to be rebased on master, there are a bunch of extra commits in here.

**Comment by neunenak** [comment]
> I think this needs to be rebased on master, there are a bunch of extra commits in here.

The new code in this PR is on top of https://github.com/casey/just/pull/1842 since that changes how attribute parsing works. I'm not sure if there's a way to tell github to omit commits from another PR in the pull request UI. In any case, https://github.com/casey/just/pull/2050/commits/eda55cc584c3eea8fa3a64d15e402a99d45655bd has the code that is actually new in this PR; I'm fine with not trying to get this merged until 1842 is merged.

File: src/recipe.rs [hunk 1]
```diff
@@ -439,6 +439,15 @@ impl<'src, D> Recipe<'src, D> {
     }
     groups
   }
+
+  pub(crate) fn docstring(&self) -> Option<&str> {
+    for attribute in &self.attributes {
+      if let Attribute::Doc { docstring } = attribute {
+        return docstring.as_ref().map(|s| s.as_str());
```
Comment (chrass) [inline src/recipe.rs]: .map(|s| s.as_str()) => .map(String::as_str())

**Comment by casey** [comment]
I'm going to cut a release tomorrow, which will be a monster one, since the features have piled up. I think this would be sweet, if you want to get it in. No worries if you don't have time though!

**Comment by neunenak** [comment]
> I'm going to cut a release tomorrow, which will be a monster one, since the features have piled up. I think this would be sweet, if you want to get it in. No worries if you don't have time though!

Yeah, it'd be great to get this in this release, although what I really care about is recipe groups so if this has to wait, not the end of the world.

**Comment by casey** [comment]
Dope!

**Comment by kranurag7** [comment]
@neunenak Thanks for working on this, I'm looking forward to using this one. 

```just
_default:
  @just --list

[doc('this is a long line
which gives usage instruction for 
test recipe')]
test:
  echo "testing"
```

When I run `just` in this directory I get the following output: 
```bash
$ just
Available recipes:
    test # this is a long line
which gives usage instruction for
test recipe
```

It would be great (in terms of readability) if can indent the extra lines as per indentation of the first line. 

```bash
$ just
Available recipes:
    test # this is a long line
           which gives usage instruction for
           test recipe
```
Is there a way I can do this now?

**Comment by simonmichael** [comment]
Is this really needed, when you can also just leave a blank line between comment and recipe ?

**Pull request merged**